### PR TITLE
add meta api

### PR DESCRIPTION
**Pull request by kqzh**
- [x] ADD HOSTS
- [x] DROP HOSTS
- [x] LIST SPACES
- [x] BALANCE

### CODE REVIEW

File: ccore/nebula/internal/driver/v3_0/meta.go [hunk 1]
```diff
@@ -1,7 +1,13 @@
 package v3_0
 
 import (
+	nerrors "github.com/vesoft-inc/nebula-http-gateway/ccore/nebula/errors"
```
Comment (veezhang) [inline ccore/nebula/internal/driver/v3_0/meta.go:4]: sort it?

File: ccore/nebula/types/driver.go [hunk 2]
```diff
@@ -65,6 +69,14 @@ type (
 		String() string
 	}
 
+	Spaces interface {
+		GetSpaceNames() []string
+	}
```
Comment (veezhang) [inline ccore/nebula/types/driver.go:74]: How about define it as:

```
Space interface {
    GetSpaceName() []string
}
Spaces []Space

func (s Spaces) GetSpaceNames() []string {
    ...
}
```

File: ccore/nebula/internal/driver/v2_5/meta.go [hunk 3]
```diff
@@ -39,3 +41,30 @@ func (c *defaultMetaClient) Close() error {
...
+
+	if err := codeErrorIfHappened(resp.Code, nil); err != nil {
+		return nil, err
+	}
+
+	return newSpacesWrapper(resp.Spaces), nil
+}
+
+func (c *defaultMetaClient) Balance(req types.BalanceReq) (types.Balancer, error) {
+	return nil, nerrors.ErrUnsupported
```
Comment (veezhang) [inline ccore/nebula/internal/driver/v2_5/meta.go:69]: Add TODO here.

File: ccore/nebula/internal/driver/v3_0/meta.go [hunk 4]
```diff
@@ -43,3 +49,114 @@ func (c *defaultMetaClient) Close() error {
 	}
 	return nil
 }
+
+func (c *defaultMetaClient) AddHosts(endpoints []string) error {
+	hostsToAdd := make([]*nthrift.HostAddr, 0, len(endpoints))
+	for _, ep := range endpoints {
```
Comment (veezhang) [inline ccore/nebula/internal/driver/v3_0/meta.go:55]: Check it at `nebula` pkg? And add an func for it?

File: ccore/nebula/types/types.go [hunk 5]
```diff
@@ -393,3 +393,17 @@ type PlanNodeBranchInfo interface {
 	String() string
 	Unwrap() interface{}
 }
+
+type BalanceCmd string
+
+const (
+	BalanceData       = BalanceCmd("Balance Data")
```
Comment (veezhang) [inline ccore/nebula/types/types.go:400]: rename it to `balanceData` ?

Comment (kqzh) [reply ccore/nebula/types/types.go:403]: driver used this, so can't private

**Review by veezhang — APPROVED** [review]
LGTM

**Pull request merged**